### PR TITLE
HDDS-7455. ClassCastException: OzoneTokenIdentifier cannot be cast to String

### PR DESCRIPTION
**Pull request by devmadhuu**
This JIRA is a bug in code while updating OM DB snapshots in Recon where framework for applying DB updates assumes storing OM DB updates in a event map with table keyType as "String" and while doing codec deserializing and type casting for DTOKENTABLE, it throws ClassCastException because for DTOKENTABLE, table key type is of "OzoneTokenIdentifier" class instead of String. 

**_Patch Fix:_**
This patch is fixed by skipping any updates or syncing of table DTOKENTABLE in Recon because Recon don't use this table for any use case as of now. If in future , DTOKENTABLE table data is needed, all events for this table updates will be saved using Object as key and new task will also retrieve using Object as key.

https://issues.apache.org/jira/browse/HDDS-7455

This patch was tested manually by creating secure cluster and tested with creating entry for DelegationToken type in dTokenTable.

### CODE REVIEW

**Comment by devmadhuu** [comment]
@sumitagrawl @dombizita @ChenSammi  pls review

**Review by dombizita — COMMENTED** [review]
thanks for the patch and the detailed comment and description regarding this decision, it looks good to me @devmadhuu

**Comment by devmadhuu** [comment]
@sadanand48  - pls review

File: hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/OMDBUpdatesHandler.java [hunk 1]
```diff
@@ -93,6 +93,18 @@ private void processEvent(int cfIndex, byte[] keyBytes, byte[]
...
+    // omdbLatestUpdateEvents defines map key as String type to store in its map
+    // and to change to Object as key will have larger impact considering all
+    // ReconOmTasks. Currently, this table is not needed to sync in Recon OM DB
+    // snapshot as this table data not being used currently in Recon.
+    // When this table data will be needed, all events for this table will be
+    // saved using Object as key and new task will also retrieve using Object
+    // as key.
+    if (OMDBDefinition.DTOKEN_TABLE.getName().equalsIgnoreCase(tableName)) {
+      return;
+    }
```
Comment (ChenSammi) [inline hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/OMDBUpdatesHandler.java:107]: Hi @devmadhuu ,  can we something like this? 

Object key = codecRegistry.asObject(keyBytes, keyType.get());
builder.setKey(key instanceof String? key : key.toString());

I'm afraid that we filter DTOKEN_TABLE this time. Next time, if there is a new table added in OM which doesn't use String as key, then this filter logic need to be updated again.

Comment (devmadhuu) [reply hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/OMDBUpdatesHandler.java]: @ChenSammi  - I have done changes as suggested. Pls re-review.

**Pull request merged**